cypress-io/cypress
# perf: remove reporter logs for collapsed tests in run mode

- Closes #25230

### Additional details

In order to reduce memory consumption, I am clearing the reporter logs for collapsed tests in run mode.

### Steps to test

Run a spec in `headed` `run` mode and verify a successful test has its reporter logs cleared and a failed test has its reporter logs retained.

### How has the user experience changed?

Before (logs retained):
<img width="686" alt="Screen Shot 2023-01-30 at 2 13 34 PM" src="https://user-images.githubusercontent.com/2002044/215596421-998c4714-00c1-4d21-aae7-d5e65fc784c9.png">

After (logs cleared):
<img width="673" alt="Screen Shot 2023-01-30 at 2 15 27 PM" src="https://user-images.githubusercontent.com/2002044/215596694-ffd8e635-5453-4e53-a115-27067c5b044f.png">

---

Running the following spec produced the following results:
```
describe('memory spec', { browser: { family: 'chromium' } }, () => {
  const text = 'x'.repeat(100000)

  for (let index = 0; index < 50; index++) {
    it(`test ${index + 1} passes`, () => {
      cy.visit('http://localhost:3500/memory')

      for (let index = 0; index < 100; index++) {
        cy.get(`#p${index}`).should('have.text', text)
      }
    })
  }
})
```

```
<html>
  <body></body>
  <script>
    for (let i = 0; i < 100; i++) {
      const el = document.createElement('p')
      el.id = 'p' + i
      el.innerHTML = 'x'.repeat(100000)
            
      document.body.appendChild(el)
    }
  </script>
</html>  
```

Before:
<img width="1396" alt="Screen Shot 2023-01-30 at 2 07 59 PM" src="https://user-images.githubusercontent.com/2002044/215595425-84ba8e07-68e0-4bcd-ba35-977b5a647ef9.png">

After:
<img width="1395" alt="Screen Shot 2023-01-30 at 2 08 27 PM" src="https://user-images.githubusercontent.com/2002044/215595495-aff7f22a-df8a-4399-b939-2f6229054658.png">

### PR Tasks


- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
